datafusion/optimizer/src/common_subexpr_eliminate.rs

```diff
@@ -43,18 +45,35 @@ const CSE_PREFIX: &str = "__common_expr";
 
 /// Identifier that represents a subexpression tree.
 ///
-/// Note that the current implementation contains:
-/// - the `Display` of an expression (a `String`) and
-/// - the identifiers of the childrens of the expression
-/// concatenated.
-///
 /// An identifier should (ideally) be able to "hash", "accumulate", "equal" and "have no
```

Comment:
```suggestion
/// This identifier is designed to be efficient and  "hash", "accumulate", "equal" and "
```

Reply:
Fixed in https://github.com/apache/datafusion/pull/10473/commits/ee61224f1a948c38ac99ae53d9d426bb69da9548.